### PR TITLE
1834792: Try to terminate rhsmd after timeout; ENT-2368

### PR DESCRIPTION
**Pull request by jirihnidek**
* BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1834792
* Don't be so agresive about terminating rhsmd
  - Wait timout defined in StatusChecker class (120 seconds)
  - After timout is reached, then check if --keep-alive
    CLI option was used and rhsm-icon is running. If any
    of this criteria is met, then rhsmd can be terminated
  - If the D-Bus API will be required again, then rhsmd will
    automatically started by dbus service

### CODE REVIEW

**Comment by jirihnidek** [comment]
Marked the PR as WIP, because I want to do more smoke testing on RHEL 7 systems

**Comment by jirihnidek** [comment]
Ready for review.

File: src/subscription_manager/scripts/rhsm_d.py [hunk 1]
```diff
@@ -96,24 +96,38 @@ def excepthook_logging(exc_type, exc_value, exc_traceback):
...
     for entitlement in sorter.valid_entitlement_certs:
         if entitlement.is_expiring():
             return True
     return False
 
 
 def pre_check_status(force_signal):
+    """
+    Pre check status of the system. Is it registered against classic. Is it registered?
+    :param force_signal: When this is not None, then this value is returned and this signal si
```
Comment (cnsnyder) [inline src/subscription_manager/scripts/rhsm_d.py:128]: I think there might have been more to this comment. "... this signal is sent"  perhaps?

File: src/subscription_manager/scripts/rhsm_d.py [hunk 2]
```diff
@@ -195,6 +241,9 @@ class StatusChecker(dbus.service.Object):
     # dbus.service.method annotation, add annotations earlier in the
     # annotation stack.
 
+    # Timeout in seconds
+    TIMEOUT = 120
```
Comment (jirihnidek) [inline src/subscription_manager/scripts/rhsm_d.py:245]: @cnsnyder @kahowell Shouldn't we use configuration option `processTimeout` from `rhsm.conf`. There is section `[rhsmd]` and it has this option. `processTimeout` is used only by bash script `rhsmd.cron` triggered by cron. Default value is 300s (5min).

File: src/subscription_manager/scripts/rhsm_d.py [hunk 3]
```diff
@@ -142,13 +162,39 @@ def check_status(force_signal):
...
+    :return:
+    """
     if checker.has_run:
         msg = "D-Bus com.redhat.SubscriptionManager.EntitlementStatus.check_status called once, exiting"
         debug(msg)
         loop.quit()
     return True
 
 
+def timout_cb(loop, checker):
```
Comment (cnsnyder) [inline src/subscription_manager/scripts/rhsm_d.py:178]: timout_cb -> timeout_cb

**Comment by cnsnyder** [comment]
Aside from the typo above, I think this is ready for merge (given a bit of functional testing as well).
Will merge after the typo is fixed :)

**Pull request merged**